File: app/src/main/java/com/mindorks/bootcamp/learndagger/di/modules/FragmentModule.java
package com.mindorks.bootcamp.learndagger.di.modules;

import android.content.Context;

import androidx.fragment.app.Fragment;

import com.mindorks.bootcamp.learndagger.di.scopes.ActivityScope;

import dagger.Module;
import dagger.Provides;

/****
 * Author: Ahmed Yehya
 * Email: ahmedyehya1992@gmail.com
 * Created on: 11/2/21
 *****/
@Module
public class FragmentModule {
    private Fragment fragment;

    public FragmentModule(Fragment fragment) {
        this.fragment = fragment;
    }

    @ActivityScope
    @Provides
    Context provideContext() {return fragment.getContext();}
}

File: app/src/main/java/com/mindorks/bootcamp/learndagger/di/modules/FragmentModule.kt
package com.mindorks.bootcamp.learndagger.di.modules

import androidx.fragment.app.Fragment
import com.mindorks.bootcamp.learndagger.di.qualifiers.ActivityContext
import dagger.Module
import dagger.Provides

/****
 * Author: Ahmed Yehya
 * Email: ahmedyehya1992@gmail.com
 * Created on: 11/10/21
 *****/
@Module
class FragmentModule (
    val fragment: Fragment
        ){
    @ActivityContext
    @Provides
    fun provideContext() = fragment.context
}